src/libraries/System.Security.Cryptography.X509Certificates/tests/CollectionTests.cs

```diff
@@ -1640,6 +1690,71 @@ private static void TestExportStore(X509ContentType ct)
             }
         }
 
+        private static void AssertPemExport(X509Certificate2Collection collection, string expectedContents)
+        {
+            Span<char> buffer = new char[expectedContents.Length];
+            int written;
+
+            // If we expect something to get written, try writing to a buffer that is too small by one
+            // and make sure it fails.
+            if (!buffer.IsEmpty)
+            {
+                Assert.False(collection.TryExportCertificatePems(buffer.Slice(1), out written), nameof(collection.TryExportCertificatePems));
+                Assert.Equal(0, written);
+            }
+
+            Assert.True(collection.TryExportCertificatePems(buffer, out written), nameof(collection.TryExportCertificatePems));
+            Assert.Equal(written, buffer.Length);
+            Assert.Equal(expectedContents, new string(buffer));
+
+            string exported = collection.ExportCertificatePems();
+            Assert.Equal(expectedContents, exported);
+        }
+
+        private static void AssertPkcs7PemExport(X509Certificate2Collection collection)
+        {
+            if (PlatformDetection.UsesMobileAppleCrypto)
+            {
+                return;
+            }
+
+            static void AssertPem(X509Certificate2Collection expected, ReadOnlySpan<char> pemActual)
```

Comment:
PKCS7 exports are interesting on different platforms. Linux / Windows appear to use DER encoding, macOS uses BER (indefinite length sequences). So we can't easily assert by content, so this just tries to make sure things round-trip correctly.